### DISCUSSION TITLE
Rendering a table of contents

### DISCUSSION THREAD

**Discussion by kevinlang** [jgm/djot#226] [General]
How would one render a TOC in djot? My first guess would be to have a list block tagged with a class or attribute like so:

```
{.toc}
*

## a heading
```

Then I guess after that I would need to run `applyFilter` or `renderHTML` and provide a `doc` "override" where I:

1. Walk the AST, finding all of the headings in order
2. Find the `.toc` node in the AST, replace it with that AST nodes that represent the TOC list

Is this the correct way to do it? As far as I can tell, if I provide an e.g., `list` override, I would not have access to the rest of the doc, so my only option is to provide a `doc` override.

Cheers,

**Comment by bdarcus** [comment]
@jgm might it be worth turning on the wiki, and encouraging people to add filter examples to it?

**Comment by jgm** [comment]
I enabled the wiki.

Yes, you'd need to walk the whole AST, collecting heading text and identifiers, and organizing these into a tree structure. Then you'd need to produce a list element in the AST, based on this tree structure, and replace the specially marked list.

**Reply by bdarcus** [reply]
Cool!

https://github.com/jgm/djot/wiki/Example-Filters
